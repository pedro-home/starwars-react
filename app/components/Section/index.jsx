import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

const Component = ({ name, children, onRef }) => {
  let props;
  if (onRef) {
    props = { ...props, ref: onRef };
  }

  return (
    <section className={classnames('section', name)} {...props}>
      {children}
    </section>
  );
};

Component.propTypes = {
  name: PropTypes.string.isRequired,
  onRef: PropTypes.func,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
};

export default Component;
