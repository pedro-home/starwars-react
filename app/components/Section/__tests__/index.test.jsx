import React from 'react';
import { render } from '@testing-library/react';

import Section from '../index';

describe('Given <Section />', () => {
  let component;
  let defaultProps;
  let text;
  let name;

  beforeAll(() => {
    text = 'Hello';
    name = 'TestSection';
    defaultProps = { children: <span>{text}</span>, name };
  });

  describe('When initializes', () => {
    beforeEach(() => {
      const { container } = render(<Section {...defaultProps} />);
      component = container;
    });

    it('Then should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    it('And should render children', () => {
      expect(component.querySelector('span').innerHTML).toEqual(text);
    });

    it('And should render class name with name', () => {
      expect(component.querySelector('.section').classList).toContainEqual(name);
    });
  });

  describe('When references component', () => {
    let ref;
    beforeEach(() => {
      const { container } = render(
        <Section
          {...defaultProps}
          onRef={elem => {
            ref = elem;
          }}
        />,
      );
      component = container;
    });

    it('Then should return DOM element', () => {
      expect(ref.classList).toContainEqual('section');
    });
  });
});
