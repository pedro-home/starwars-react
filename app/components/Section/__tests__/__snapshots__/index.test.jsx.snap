// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <Section /> When initializes Then should match the snapshot 1`] = `
<div>
  <section
    class="section TestSection"
  >
    <span>
      Hello
    </span>
  </section>
</div>
`;
